File: lista3/zad2/TestDH.java

public class TestDH {
    public static void main(String[] args) {
        GaloisField.setPrimeNumber(1234567891);
        //DHSetup<GaloisField> dhs = new DHSetup<GaloisField>(GaloisField.getPrimeNumber(), new GaloisField(0));
        DHSetup<GaloisField> dhs = new DHSetup<GaloisField>(GaloisField.getPrimeNumber(), new GaloisField(0));
        DHUser<GaloisField> user1 = new DHUser<GaloisField>(dhs, GaloisField.getPrimeNumber());
        DHUser<GaloisField> user2 = new DHUser<GaloisField>(dhs, GaloisField.getPrimeNumber());
        try {
            user1.setKey(user2.getPublicKey());
            GaloisField e = user1.encrypt(new GaloisField(111436077));
            user2.setKey(user1.getPublicKey());
            System.out.print("e: ");
            user2.decrypt(e).println();
            e = user2.encrypt(new GaloisField(841536281));
            System.out.print("e: ");
            user1.decrypt(e).println();
        } catch (Exception ex) {
            System.out.println("Error: " + ex.getMessage());
        }
    }
}

File: lista3/zad3/testDH.py

from galoisField import *
from diffieHellman import *
from DHUser import *
import pytest

# Program to test the implementation of Diffie-Hellman algorithm using objects of class GaloisField.

GaloisField.setPrimeNumber(1234567891)
dhs = DHSetup(GaloisField.getPrimeNumber(), GaloisField(0))

user1 = User(dhs, GaloisField.getPrimeNumber())
user2 = User(dhs, GaloisField.getPrimeNumber())

# attempt to encrypt without the key set
with pytest.raises(Exception):
    e1 = user1.encrypt(GaloisField(1114367))
with pytest.raises(Exception):
    d2 = user2.encrypt(GaloisField(1114367))

user1.setKey(user2.getPublicKey())
e1 = user1.encrypt(GaloisField(1114367))

# attempt to decrypt without the key set
with pytest.raises(Exception):
    d2 = user2.decrypt(e1)

user2.setKey(user1.getPublicKey())
assert user2.decrypt(e1) == 1114367

e1 = user2.encrypt(GaloisField(0))
assert user1.decrypt(e1) == 0

print("All tests passed.")